osquery/config/config.cpp

```diff
@@ -531,6 +554,58 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+Expected<ConfigMap, Config::RestoreConfigError> Config::restoreConfigBackup() {
+  if (!FLAGS_config_enable_backup) {
+    return createError(Config::RestoreConfigError::BackupDisabled,
+                       "Config persistense is disabled.");
+  }
+
+  VLOG(1) << "Restoring backed up config from the database";
+  std::vector<std::string> keys;
+  ConfigMap config;
+
+  WriteLock lock(config_backup_mutex_);
+  scanDatabaseKeys(kPersistentSettings, keys, kConfigPersistencePrefix);
+
+  for (const auto& key : keys) {
+    std::string value;
+    Status status = getDatabaseValue(kPersistentSettings, key, value);
+    if (!status.ok()) {
+      VLOG(1)
```

Comment:
From my point of view this should be a fatal error (since we don't have the concept of fatal yet, at least a `LOG(ERROR)`) since basically you fail to retrieve some of the configuration either because of a DB error or because something concurrently removed the key you're expecting.